src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_keychain.cpp

```diff
@@ -208,6 +208,95 @@ static OSStatus DeleteInKeychain(CFTypeRef needle, SecKeychainRef haystack)
 }
 
 extern "C" int32_t
+AppleCryptoNative_X509StoreAddCertificate(CFTypeRef certOrIdentity, SecKeychainRef keychain, int32_t* pOSStatus)
+{
+    if (certOrIdentity == nullptr || keychain == nullptr)
+        return -1;
+
+    SecCertificateRef cert = nullptr;
+    SecKeyRef privateKey = nullptr;
+
+    auto inputType = CFGetTypeID(certOrIdentity);
+    OSStatus status = noErr;
+
+    if (inputType == SecCertificateGetTypeID())
+    {
+        cert = reinterpret_cast<SecCertificateRef>(const_cast<void*>(certOrIdentity));
+        CFRetain(cert);
+    }
+    else if (inputType == SecIdentityGetTypeID())
+    {
+        SecIdentityRef identity = reinterpret_cast<SecIdentityRef>(const_cast<void*>(certOrIdentity));
+        status = SecIdentityCopyCertificate(identity, &cert);
+
+        if (status == noErr)
+        {
+            status = SecIdentityCopyPrivateKey(identity, &privateKey);
+        }
+    }
+    else
+    {
+        return -1;
```

Comment:
It looks like pOSStatus is declared as out in the C# code... we should make sure we've initialized it on all return paths.